src/main/java/frc/robot/subsystems/drive/Drive.java

```diff
@@ -157,18 +160,53 @@ public void arcadeDrive(double throttle, double turnModifier){
     differentialDrive.arcadeDrive(throttle, turnModifier);
   }
 
+  /**
+   * Executes curvatureDrive with closed loop control. Needs to be called
+   * periodically by the command to update the closed loop. 
+   * 
+   * The closed loop works on the difference between the desired curvature
+   * and the measured curvature, using proportional gain only.
+   * 
+   * @param throttle - [-1.0, 1.0]. Positive is forward
+   * @param curvature - [-1.0, 1.0]. Positive is turning right
+   */
   public void curvatureDrive(double throttle, double curvature) {
-    if(throttle != 0.0 && curvature == 0.0 && !gyroDisabled) { //driving straight and no turn
-      if(this.desiredAngle == Integer.MAX_VALUE) { //means robot just started driving straight
-        this.desiredAngle = gyroscope.getGyroAngle(); 
-      }
-      curvature = this.getAngularError(desiredAngle) * P_VALUE; 
-      this.setLeftPower(throttle - curvature);
-      this.setRightPower(throttle + curvature);
-    }
-    else { // when robot isn't driving straight
-      this.desiredAngle = Integer.MAX_VALUE; //if turn is greater than 0 or if robot is still
-      differentialDrive.curvatureDrive(-throttle, curvature, (Math.abs(throttle) < QUICK_TURN_THROTTLE_DEADZONE));
+
+    if(Math.abs(throttle) < QUICK_TURN_THROTTLE_DEADZONE){
+      // If throttle is less than the deadzone, treat inputs as arcade
+      this.arcadeDrive(throttle, curvature);
+    }else{
+      // Otherwise, treat inputs as curvature
+
+      // Calculate ideal left and right speeds based on curvature
+      WheelSpeeds desiredSpeeds = DifferentialDrive.curvatureDriveIK(throttle, curvature, false);
```

Comment:
What do you think about moving the quick turn condition here?
```suggestion
      WheelSpeeds desiredSpeeds = DifferentialDrive.curvatureDriveIK(throttle, curvature, Math.abs(throttle) < QUICK_TURN_THROTTLE_DEADZONE);
```

Reply:
With this, we could also get rid of `arcadeDrive()`

Reply:
This would run arcade drive through closed loop on curvature, and if linear velocity is at or near 0 curvature would be at or near infinity. This would be caught by the floors on ideal and measured linear velocity used in calculation, so we can implement it, but I'd want to label a few things better.

Reply:
Sorry I should have been more clear, we would also wrap the closed loop math with this conditional to avoid the closed loop calculations when quick turning.